src/managers/ApplicationCommandManager.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const BaseManager = require('./BaseManager');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const Collection = require('../util/Collection');
+
+/**
+ * Manages API methods for application commands and stores their cache.
+ * @extends {BaseManager}
+ */
+class ApplicationCommandManager extends BaseManager {
+  constructor(client, iterable) {
+    super(client, iterable, ApplicationCommand);
+  }
+
+  /**
+   * The cache of this manager
+   * @type {Collection<Snowflake, ApplicationCommand>}
+   * @name ApplicationCommandManager#cache
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { extras: [this.guild] });
+  }
+
+  /**
+   * The APIRouter path to the commands
+   * @type {Object}
+   * @readonly
+   * @private
+   */
+  get commandPath() {
+    let path = this.client.api.applications(this.client.application.id);
+    if (this.guild) path = path.guilds(this.guild.id);
+    return path.commands;
+  }
+
+  /**
+   * Data that resolves to give an ApplicationCommand object. This can be:
+   * * An ApplicationCommand object
+   * * A Snowflake
+   * @typedef {ApplicationCommand|Snowflake} ApplicationCommandResolvable
+   */
+
+  /**
+   * Obtains one or multiple application commands from Discord, or the cache if it's already available.
+   * @param {Snowflake} [id] ID of the application command
+   * @param {boolean} [cache=true] Whether to cache the new application commands if they weren't already
+   * @param {boolean} [force=false] Whether to skip the cache check and request the API
+   * @returns {Promise<ApplicationCommand|Collection<Snowflake, ApplicationCommand>>}
+   */
+  async fetch(id, cache = true, force = false) {
+    if (id) {
+      if (!force) {
+        const existing = this.cache.get(id);
+        if (existing) return existing;
+      }
+      const command = await this.commandPath(id).get();
+      return this.add(command, cache);
+    }
+
+    const data = await this.commandPath.get();
+    return data.reduce((coll, command) => coll.set(command.id, this.add(command, cache)), new Collection());
+  }
+
+  /**
+   * Creates an application command.
+   * @param {ApplicationCommandData} command The command
+   * @returns {Promise<ApplicationCommand>}
+   */
+  async create(command) {
```

Comment:
Functions such as `ApplicationCommandManager#edit` and `ApplicationCommandManager#delete` would be quite lovely, similar to #4869 and #4872. I would suggest an upsert function however from reading the docs posting [already does that](https://discord.com/developers/docs/interactions/slash-commands#create-global-application-command).